publicode/rules/dirigeant.yaml

```diff
@@ -648,6 +666,101 @@ dirigeant . indépendant . cotisations et contributions . cotisations . déducti
+
+dirigeant . indépendant . contrats madelin . part non-déductible fiscalement:
+  titre: Part de la cotisation à contrat Madelin qui n'est pas déductible fiscalement
+  formule: montant - part déductible fiscalement
+
+dirigeant . indépendant . contrats madelin . mutuelle:
+  titre: Contrat Madelin mutuelle
+
+dirigeant . indépendant . contrats madelin . mutuelle . montant:
+  titre: Souscription à un contrat de mutuelle Madelin
+  unité par défaut: €/an
+  question: Quel est le montant que vous versez à un contrat de mutuelle Madelin ?
+  description: |
+    Si vous cotisez au titre d'un contrat de mutuelle de type loi Madelin,
+    vous pouvez déduire une partie de ces cotisations des bénéfices
+    imposables que vous déclarez pour votre activité non salariée.
+  références:
+    Fiche impôts: https://www.impots.gouv.fr/portail/particulier/questions/je-cotise-un-contrat-madelin-quel-est-mon-avantage-fiscal
+    Bofip (contrats d'assurance de groupe): https://bofip.impots.gouv.fr/bofip/4639-PGP.html
+    Article de loi: https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000029042287&cidTexte=LEGITEXT000006069577&dateTexte=20140530&fastReqId=1900907951&nbResultRech=1
+  par défaut: 0
+
+dirigeant . indépendant . contrats madelin . mutuelle . plafond:
+  unité par défaut: €/an
+  formule:
+    le minimum de:
+      - somme:
+        - produit:
+            assiette: revenu professionnel
+            taux: 3.75%
+        - produit:
+            assiette: plafond sécurité sociale temps plein
+            taux: 7%
+      - produit:
+          assiette: 8 * plafond sécurité sociale temps plein
+          taux: 3%
+  références:
+    Réassurez-moi: https://reassurez-moi.fr/guide/pro/tns/plafond#le_plafond_de_deduction_madelin_pour_une_mutuelle_santenbsp
+
+dirigeant . indépendant . contrats madelin . retraite:
+  titre: Contrat Madelin retraite
+
+dirigeant . indépendant . contrats madelin . retraite . montant:
+  titre: Souscription à une retraite Madelin
+  unité par défaut: €/an
+  question: Quel est le montant que vous versez à votre contrat Madelin retraite ?
+  description: |
+    Si vous cotisez au titre d'un contrat retraite de type loi Madelin,
+    vous pouvez déduire une partie de ces cotisations des bénéfices
+    imposables que vous déclarez pour votre activité non salariée.
+  références:
+    Fiche impôts: https://www.impots.gouv.fr/portail/particulier/questions/je-cotise-un-contrat-madelin-quel-est-mon-avantage-fiscal
+    Bofip (contrats d'assurance de groupe): https://bofip.impots.gouv.fr/bofip/4639-PGP.html
+    Article de loi: https://www.legifrance.gouv.fr/affichCodeArticle.do?idArticle=LEGIARTI000029042287&cidTexte=LEGITEXT000006069577&dateTexte=20140530&fastReqId=1900907951&nbResultRech=1
+  par défaut: 0
+
+dirigeant . indépendant . contrats madelin . retraite . plafond:
+  unité par défaut: €/an
+  formule:
+    le maximum de:
+      - barème:
+          assiette: revenu professionnel
```

Comment:
@johangirod apparemment d'après l'expert il s'agit de mettre ici `résultat fiscal`, qui lui-meme depend du plafond.

A priori ça devrait faire un cycle, mais il semblerait que ça fonctionne quand même. Je vais creuser la question en regardant plus en détail l'issue Cycles.

Ceci dit ma question est plus d'ordre logique: semble-t-il cohérent d'avoir un plafond qui dépend du résultat fiscal? Sachant que:

- dans la construction de règles actuelle, on a la formule `assiette des cotisations + contrats madelin . montant` (et `assiette des cotisations == revenu professionnel`)
- dans la decomposition "topline" (a partir du CA), j'ai: `résultat fiscal = revenu net de cotisations - cotisations et contributions + reintegrations`, avec reintegrations qui comprend la partie non-deductible des contrats madelin.

Le fait que les deux décompositions nous donnent qqch de cyclique me semble pas très bon. Mais peut-être est-ce un autre cas de figure du "Zen of CSG": c'est un cycle dans le formalisme de publicodes, mais l'équation peut, elle, bien être résolue (notamment via la méthode numérique appliquée).

A mon avis tant que ça marche, il faut faire la modif proposée par l'expert à moins que tu aies une contre-indication @johangirod . Ca sera un parfait cas de figure "pathologique" pour moi à traiter pour l'histoire des cycles.

Reply:
Je suis arrivé à la même conclusion que toi lorsque j'ai reçu la réponse de l'expert : ça va faire un cycle. Le fait que ça fonctionne quand même viens, je pense, de la redéfinition du pivot "revenu professionnel" au résultat fiscal. Il y a un cycle, mais il est résolu par l'inversion.

La question "philosophique" sous-jacente est plus dérangeante. Pour moi, un cycle dans la détermination séquentielle des différents résultats (dans l'ordre : comptable, fiscal, social) amène à deux issues possibles : 

- L'expert a raison sur le fond, mais on s'aperçoit que les logiciels de comptabilité procèdent à une simplification (en calculant le plafond sur une valeur intermédiaire)
- L'expert a tort et le revenu professionnel est bien celui qui sert de base aux cotisations

Comment trancher entre les deux ? Dans l'état actuel de nos connaissances, je partirai sur la première option. Quitte à revenir dessus si des découvertes postérieurs nous amène à douter de cette solution. 

En tout état de cause, cette reflexion mérite de figurer sous une forme ou une autre dans la base de règle. Peut-être dans l'attribut `note` du revenu professionel ?

Reply:
Après discussion on voit que c'est bien le résultat fiscal qu'il faut utiliser, cf.: 

![DD IR B_GC_2011_03_25_11image2](https://user-images.githubusercontent.com/1884912/78548916-d0b45100-7801-11ea-903c-ac3e2499bed9.jpg)